pkg/proxy/proxyinfo.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"istio.io/istio/pilot/pkg/xds"
+	"istio.io/istio/pkg/kube"
+	istioVersion "istio.io/pkg/version"
+)
+
+type sidecarSyncStatus struct {
+	// nolint: structcheck, unused
+	pilot string
+	xds.SyncStatus
+}
+
+func GetProxyInfo(kubeconfig, configContext, revision, istioNamespace string) (*[]istioVersion.ProxyInfo, error) {
+	kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), revision)
```

Comment:
can we pass in client?

Reply:
we can, but what is the advantage? Seems like we need to initialize the client and config every time before calling? And actually this is istio kube client wrapper not the native kube client

Reply:
just as a code hygiene thing. you don't want to be creating clients in downstream functions like this - those things should be at the top level and passed around when necessary. for example - if you need to do something else involving a client in uninstall in the future, you'd have to refactor this, or the author will likely create the client again. this is exactly what happened with other istioctl manifest commands and has been cleaned up since - have a look at the pattern there.